### DISCUSSION TITLE
Is it documented somewhere that not all underlying node-postgres properties are exposed as config options?

### DISCUSSION THREAD

**Discussion by angusryer** [vitaly-t/pg-promise#903] [Q&A]
I was troubleshooting a failing connecting to my Google Postgres instance and found that I need to include a `host` property in my SSL configuration. The `ISSLConfig` interface doesn't have this property.

Since documenting how to connect with each and every service out there isn't likely, I'm posting this as reference for and to be improved by others.

---

This is for those that require connecting to their GCP Cloud SQL Postgres instance via SSL certs instead of Cloud Proxy (e.g. for serverless environments).

In certain libraries that use `node-postgres` under the hood (`pg-promise`, for instance), the `ssl` object used to configure the DB connection doesn't expose all the underlying `node` properties by default.

*If you created your GCP PostgreSQL instance before January 2023*, the `host` property is required, in the following format, alongside the `key`, `cert` and `ca` properties, otherwise the `ERR_TLS_CERT_ALTNAME_INVALID` error is thrown.

``` javascript
const ssl = {
   ca: <your CA>,
   key: <your key>,
   cert: <your cert>,
   host: "your-google-cloud-project:your-postgres-instance"
}
```

Note that the `host` syntax is not the `project:region:instance` format that you'll find the GCP SQL Cloud Console. It's just `project:instance`.

I can't speak to how the implementation would be for instances created _after_ January 2023, but there's talk of changes in `node` itself and changes in Google's approach as well.

The following thread has been my reference and talks about implementations of instances after January 2023 as well: https://github.com/brianc/node-postgres-docs/issues/79#issuecomment-1553759056

Anything to add or clarify is welcome!

**Comment by vitaly-t** [comment]
Type Script declarations in this library expose what the [underlying driver](https://github.com/brianc/node-postgres) supports. And if `host` isn't supported, then it's not here either. If something is amiss here, you can add a PR for it 😉
